controllers/ip_controller.go

```diff
@@ -231,6 +254,81 @@ func getCurrentIP(sources []string) (string, error) {
 	return currentIP, nil
 }
 
+// getStaticIP returns the IP gathered from the staticIPSource
+func (r *IPReconciler) getStaticIP(ctx context.Context, instance *cfv1beta1.IP) (string, error) {
+	_, err := url.Parse(instance.Spec.StaticIPSource.URL)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse URL %s: %s", instance.Spec.StaticIPSource.URL, err)
+	}
+
+	httpClient := &http.Client{}
+	req, err := http.NewRequest(instance.Spec.StaticIPSource.RequestMethod, instance.Spec.StaticIPSource.URL,
+		io.Reader(bytes.NewBuffer([]byte(instance.Spec.StaticIPSource.RequestBody))))
+
+	for key, value := range instance.Spec.StaticIPSource.RequestHeaders {
+		req.Header.Add(key, value)
+	}
+
+	if instance.Spec.StaticIPSource.RequestHeadersSecretRef.Name != "" {
+		secret := &v1.Secret{}
+		err := r.Get(ctx, types.NamespacedName{Name: instance.Spec.StaticIPSource.RequestHeadersSecretRef.Name,
+			Namespace: instance.Spec.StaticIPSource.RequestHeadersSecretRef.Namespace}, secret)
+		if err != nil {
+			return "", fmt.Errorf("failed to get secret %s: %s", instance.Spec.StaticIPSource.RequestHeadersSecretRef.Name, err)
+		}
+		for key, value := range secret.Data {
+			req.Header.Add(key, string(value))
+		}
+	}
+
+	httpClient.Timeout = time.Second * 30
+	req.Header.Add("User-Agent", "cloudflare-operator")
+
+	resp, err := httpClient.Do(req)
+	if err != nil {
+		return "", fmt.Errorf("failed to get IP from %s: %s", instance.Spec.StaticIPSource.URL, err)
+	}
+	if resp.StatusCode != 200 {
+		return "", fmt.Errorf("failed to get IP from %s: %s", instance.Spec.StaticIPSource.URL, resp.Status)
+	}
+	ip, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", fmt.Errorf("failed to get IP from %s: %s", instance.Spec.StaticIPSource.URL, err)
+	}
+
+	var extractedIP string
+	switch instance.Spec.StaticIPSource.ResponseType {
+	case "json":
+		var jsonResponse map[string]interface{}
+		err := json.Unmarshal(ip, &jsonResponse)
+		if err != nil {
+			return "", fmt.Errorf("failed to get IP from %s: %s", instance.Spec.StaticIPSource.URL, err)
+		}
+		j := jsonpath.New("jsonpath")
+		buf := new(bytes.Buffer)
+		if err := j.Parse(instance.Spec.StaticIPSource.ResponseJSONPath); err != nil {
+			return "", fmt.Errorf("failed to parse jsonpath %s: %s", instance.Spec.StaticIPSource.ResponseJSONPath, err)
+		}
+		if err := j.Execute(buf, jsonResponse); err != nil {
+			return "", fmt.Errorf("failed to extract IP from %s: %s", instance.Spec.StaticIPSource.URL, err)
+		}
+		extractedIP = strings.TrimSpace(buf.String())
+	case "text":
+		extractedIP := string(ip)
+		if instance.Spec.StaticIPSource.ResponseTextRegex == "" {
+			return extractedIP, nil
+		}
+		re := regexp.MustCompile(instance.Spec.StaticIPSource.ResponseTextRegex)
+		match := re.FindStringSubmatch(extractedIP)
+		if len(match) == 0 {
+			return "", fmt.Errorf("failed to extract IP from %s: %s", instance.Spec.StaticIPSource.URL, extractedIP)
+		}
+		extractedIP = strings.TrimSpace(match[1])
```

Comment:
use last match and not second
```suggestion
		extractedIP = strings.TrimSpace(match[len(match)-1])
```